### DISCUSSION TITLE
Get `flux` Python package from path to `flux` executable

### DISCUSSION THREAD

**Discussion by jameshcorbett** [flux-framework/flux-core#3665] [Q&A] [answered]
As part of my Parsl work I'm trying to drive a Flux instance remotely. If I could import the ``flux`` python package into my Python environment, I would be good to go. I can [fetch the URI from the instance](https://github.com/flux-framework/flux-core/issues/3604#issuecomment-823775023) and connect to it that way. However, I don't know how to get the Python package. (I'm used to just having the package available because an encapsulating Flux instance has set `PYTHONPATH`.) Is there a reliable way to find the path to the Python package, given a path to the `flux` executable or some other reference point? Either by querying the executable or by following a certain predefined path.

Worst case I can fetch both the URI and the path to the package from inside the instance, by checking `flux.__file__`.

**Comment by dongahn** [comment, marked as answer]
```console
ahn1@docker-desktop:/usr/src$ flux env | grep PYTHONPATH
export PYTHONPATH="/usr/lib/flux/python3.6"
```

**Comment by dongahn** [comment]
How `flux` constructs `PYTHONPATH` is also documented in https://flux-framework.readthedocs.io/projects/flux-core/en/latest/man1/flux.html
